utils/build.ps1

```diff
@@ -1465,6 +1471,59 @@ function Build-Compilers() {
   }
 }
 
+# Reference: https://github.com/microsoft/mimalloc/tree/dev/bin#minject
+# TODO: Add ARM64
+function Build-Mimalloc() {
+  [CmdletBinding(PositionalBinding = $false)]
+  param
+  (
+    [Parameter(Position = 0, Mandatory = $true)]
+    [hashtable]$Arch
+  )
+
+  if ($Arch -eq $ArchX64) {
+    $Args = @()
+    Isolate-EnvVars {
+      Invoke-VsDevShell $Arch
+      # Avoid hard-coding the VC tools version number
+      $VCRedistDir = (Get-ChildItem "${env:VCToolsRedistDir}\$($HostArch.ShortName)" -Filter "Microsoft.VC*.CRT").FullName
+      if ($VCRedistDir) {
+        $Args += "-p:VCRedistDir=$VCRedistDir\"
+      }
+    }
+    $Args += "$SourceCache\mimalloc\ide\vs2022\mimalloc.sln"
+    $Args += "-p:Configuration=Release"
+    $Args += "-p:ProductArchitecture=$($Arch.VSName)"
+    Invoke-Program $msbuild @Args
+    $Dest = "$($Arch.ToolchainInstallRoot)\usr\bin"
+    Copy-Item -Path "$SourceCache\mimalloc\out\msvc-$($Arch.ShortName)\Release\mimalloc-override.dll" `
+      -Destination "$Dest"
+    Copy-Item -Path "$SourceCache\mimalloc\out\msvc-$($Arch.ShortName)\Release\mimalloc-redirect.dll" `
+      -Destination "$Dest"
+    $MimallocExecutables = @("swift.exe","swiftc.exe","swift-driver.exe","swift-frontend.exe")
+    $MimallocExecutables += @("clang.exe","clang++.exe","clang-cl.exe")
+    $MimallocExecutables += @("lld.exe","lld-link.exe","ld.lld.exe","ld64.lld.exe")
+    foreach ($Exe in $MimallocExecutables) {
+      $ExePath = [IO.Path]::Combine($Dest, $Exe)
+      # Binary-patch in place
+      $Args = @()
+      $Args += "-f"
+      $Args += "-i"
+      $Args += "-v"
+      $Args += $ExePath
+      Invoke-Program "$SourceCache\mimalloc\bin\minject" @Args
+      # Log the import table
+      $Args = @()
+      $Args += "-l"
+      $Args += $ExePath
+      Invoke-Program "$SourceCache\mimalloc\bin\minject" @Args
+      dir "$ExePath"
```

Comment:
Debugging left overs?

Reply:
I mean to keep it so that we can verify that the import table is updated in the toolchain build log